p2719.md

```diff
@@ -848,3 +951,56 @@ This proposal does not have any impact on the library, since this only tweaks th
+type being allocated. For P2719 we are not adjusting the semantics of coroutine allocations to
+resolve type aware new or delete. The coroutine specification will need to be updated in the future to
+determine what type will be passed, and to include that type in the allocation function resolution
+(similar to what is achievable through `decltype(lambda)`).
+
+# Future work
+
+This proposal has exposed a number of gaps in the current language specification, that would ideally
+be addressed in future proposals.
+
+## Coroutine frame types
+
+Coroutines explicitly resolve `operator new` and `operator delete` independently of other resolution
+paths, but even when that resolution is performed they do not provide any type information about the
+allocation being performed. From discussions with multiple vendors, the exact content of the promise
+frame being allocated is not determined until _after_ front end and semantic analysis has occured: the
+content of the type being allocated is determined by the codegenerator.
+
+Given those semantics it seems implausible that we will ever be able to provide a type that
+accurately reflects the exact type being allocated, however for type aware allocators being able to
+distinguish a coroutine frame from an opaque bucket of bits is critical, as coroutine frames are
+analagous to stack frames and can contain critical data - code pointers, post validation spills,
+etc - as well as being subject to all the historical attacks applied to any other stack frame.
+
+At minimum some degree of tag type will be needed, but it seems likely that at least some metadata
+and queries should be possible prior to codegen, so the exact design of the type requires discussion
+that is outside the scope of this proposal.
+
+## constexpr allocation functions
+
+In order to support functional requirements of a system allocator, this proposal adds wording to
+special case globally defined type aware allocation functions that are filling the space usually
+reserved for the _global replaceable allocation functions_. This wording is necessary as the current
+specification language explicitly precludes any _new-_ or _delete-expression_ where the resolved
+operator is not explicitly a globally replaceable allocation function.
+
+This restriction is not necessary as constexpr `new` and `delete` are both implementable via the
+constexpr `std::allocator` API, or via direct calls to `::operator new(...)` and
+`::operator delete(...)`. As a result of this restriction no class, or class hierarchy, may be used
+during constant expression evaluation, because the mere existence of the class scoped allocation
+function precludes lookup from ever reaching the global scope. This means that any scenarios where
+such a type must be used in a constant evaluation context must be forced to use a global operators,
+potentially leading to security or correctness errors, or must use copious `if consteval` blocks
+through out otherwise clean code.
```

Comment:
I think this is extremely interesting, however it's also not really related to this proposal. I don't think it's really useful to add this information here. You should instead note this down and come back with proposals to address this in the future (if you think it's worth it).

```suggestion

```